### PR TITLE
NSTextField extension for observing all text changes

### PR DESCRIPTION
**Pull request by skyylex**
rac_textSignal covers only UI changes, in some cases is useful to observe all changes no matter how they were generated

### CODE REVIEW

**Comment by neilpa** [comment]
Thanks for the PR and sorry for the delay here. However, this isn't something I want to add. I'm trying to move away the more invasive Obj-C RAC methods like `rac_signalForSelector` (c.f. #18).

**Comment by neilpa** [comment]
I'd be fine limiting this to a wrapper around `NSTextField.rac_textSignal` though.

**Comment by skyylex** [comment]
It's possible to create a Swift only implementation of the rac_textSignal just because it's based on the NSNotificationCenter observing - https://github.com/ReactiveCocoa/ReactiveCocoa/blob/master/ReactiveCocoa/Objective-C/NSText%2BRACSignalSupport.m

**Comment by skyylex** [comment]
Something like this:

```
public var rex_textSignal: SignalProducer<String, NoError> {
    let notificationCenter = NSNotificationCenter.defaultCenter()
    let notificationsProducer = notificationCenter.rac_notifications(name: NSControlTextDidChangeNotification, object: self)
    return notificationsProducer
        |> filter { notification in (notification.object as? NSTextField) == self }
        |> map { notification in (notification.object as! NSTextField).stringValue }
}
```

**Comment by neilpa** [comment]
I'd be fine with that implementation as well. My main concern was with avoiding `rac_signalForSelector`.

**Comment by skyylex** [comment]
I left only Swift implementation. Please take a look.

File: Source/AppKit/NSTextField.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import ReactiveCocoa
+import AppKit
+
+extension NSTextField {
+    /// only changes from UI will be produced here
+    public var rex_textSignal: SignalProducer<String, NoError> {
+        let notificationCenter = NSNotificationCenter.defaultCenter()
+        let notificationsProducer = notificationCenter.rac_notifications(name: NSControlTextDidChangeNotification, object: self)
+        return notificationsProducer
+            |> filter { notification in (notification.object as? NSTextField) == self }
```
Comment (neilpa) [inline Source/AppKit/NSTextField.swift]: Is this `filter` necessary? Does passing `self` in the call to `rac_notifications` not already scope the notifications to just this object?

File: Source/AppKit/NSTextField.swift [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import Foundation
+import ReactiveCocoa
+import AppKit
+
+extension NSTextField {
+    /// only changes from UI will be produced here
+    public var rex_textSignal: SignalProducer<String, NoError> {
+        let notificationCenter = NSNotificationCenter.defaultCenter()
+        let notificationsProducer = notificationCenter.rac_notifications(name: NSControlTextDidChangeNotification, object: self)
+        return notificationsProducer
```
Comment (neilpa) [inline Source/AppKit/NSTextField.swift]: Can you remove this local and just return the signal chain directly.

**Comment by neilpa** [comment]
Awesome, just a couple of comments.

**Comment by skyylex** [comment]
Updated according to the comments.

**Comment by neilpa** [comment]
Thanks :fireworks:

**Pull request merged**